### PR TITLE
[ITSM][Phone] add oauthCredentials

### PR DESCRIPTION
**Pull request by xieofxie**
### Purpose
*What is the context of this pull request? Why is it being done?*

Follow https://github.com/microsoft/botframework-skills/pull/149

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

### CODE REVIEW

**Comment by lzc850612** [comment]
Do these two skills also use AAD oauth?

**Comment by xieofxie** [comment]
> Do these two skills also use AAD oauth?

Phone uses, ITSM not. Does this approach support 3rd party oauth too?

**Comment by lzc850612** [comment]
> > Do these two skills also use AAD oauth?
> 
> Phone uses, ITSM not. Does this approach support 3rd party oauth too?

Do you mean 3rd party skill? it wouldn't support 3rd party skill as it'll require the skill has a static setting on a particular VA's oauth credentials settings. for 3rd party skill SSO there's the SSO approach: https://github.com/microsoft/botframework-solutions/pull/3243. i'm working on documentation for it as well.

yeah we should add this setting for all the skills that require oauth connections. are these two skills the only ones that require oauth to work except for calendar, email and todo?

**Comment by xieofxie** [comment]
> > > Do these two skills also use AAD oauth?
> > 
> > 
> > Phone uses, ITSM not. Does this approach support 3rd party oauth too?
> 
> Do you mean 3rd party skill? it wouldn't support 3rd party skill as it'll require the skill has a static setting on a particular VA's oauth credentials settings. for 3rd party skill SSO there's the SSO approach: [microsoft/botframework-solutions#3243](https://github.com/microsoft/botframework-solutions/pull/3243). i'm working on documentation for it as well.
> 
> yeah we should add this setting for all the skills that require oauth connections. are these two skills the only ones that require oauth to work except for calendar, email and todo?

No, I mean oauth that are not AAD.

Yes, these two are the only remaining ones

File: skills/csharp/experimental/itsmskill/Dialogs/SkillDialogBase.cs [hunk 1]
```diff
@@ -51,7 +52,15 @@ public class SkillDialogBase : ComponentDialog
                 throw new Exception("You must configure an authentication connection before using this component.");
             }
 
-            AddDialog(new MultiProviderAuthDialog(Settings.OAuthConnections));
+            AppCredentials oauthCredentials = null;
+            if (Settings.OAuthCredentials != null &&
+                !string.IsNullOrWhiteSpace(Settings.OAuthCredentials.MicrosoftAppId) &&
```
Comment (sidmore) [inline skills/csharp/experimental/itsmskill/Dialogs/SkillDialogBase.cs:57]: Empty string?

Comment (lzc850612) [reply skills/csharp/experimental/itsmskill/Dialogs/SkillDialogBase.cs:57]: white space is better :) it includes check for empty string

**Pull request merged**